02_activities/assignments/assignment_2.ipynb

```diff
@@ -1,339 +1,445 @@
+    "## 2. Data Summarization Function: `patient_summary`\n",
+    "\n",
+    "Your next step is to create a function named `patient_summary` that will compute summary statistics for each patient's data over a 40-day period.\n",
+    "\n",
+    "**Function Specifications**:\n",
+    "- **Function Name**: `patient_summary`\n",
+    "- **Parameters**:\n",
+    "  1. `file_path`: A string representing the path to the CSV file containing the patient data.\n",
+    "  2. `operation`: A string specifying the type of summary operation to perform. Acceptable values are \"mean\", \"max\", or \"min\". This will determine whether the function calculates the average, maximum, or minimum number of flare-ups for each patient over the 40 days.\n",
+    "\n",
+    "**Expected Behavior**:\n",
+    "- Your function should read the data from the file at `file_path`.\n",
+    "- Perform the specified `operation` (mean, max, or min) to summarize the flare-ups data for each of the 60 patients.\n",
+    "- Return an array with 60 elements, each element being the result of the summary operation for a corresponding patient.\n",
+    "\n",
+    "**Expected Output**:\n",
+    "- The output should be an array with a length of 60, aligning with the number of patients in the study.\n",
+    "\n",
+    "**Hints for Implementation**:\n",
+    "1. **Utilizing NumPy**: For efficient data manipulation and computation, consider using NumPy, as discussed in the `10_numpy` slides.\n",
+    "2. **Output Shape**: Ensure that the shape of your output data matches the number of patients, which is 60."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "82-bk4CBB1w4"
+   },
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "\n",
+    "def patient_summary(file_path, operation):\n",
+    "    # load the data from the file\n",
+    "    data = np.loadtxt(fname=file_path, delimiter=',')\n",
+    "    ax = 1  # this specifies that the operation should be done for each row (patient)\n",
+    "\n",
+    "    # implement the specific operation based on the 'operation' argument\n",
+    "    if operation == 'mean':\n",
+    "        # YOUR CODE HERE: calculate the mean (average) number of flare-ups for each patient\n",
+    "        result = np.mean(data, axis=1)\n",
+    "\n",
+    "    elif operation == 'max':\n",
+    "        # YOUR CODE HERE: calculate the maximum number of flare-ups experienced by each patient\n",
+    "        result = np.max(data, axis=1)\n",
+    "\n",
+    "    elif operation == 'min':\n",
+    "        # YOUR CODE HERE: calculate the minimum number of flare-ups experienced by each patient\n",
+    "         result = np.min(data, axis=1)\n",
+    "\n",
+    "    else:\n",
+    "        # if the operation is not one of the expected values, raise an error\n",
+    "        raise ValueError(\"Invalid operation. Please choose 'mean', 'max', or 'min'.\")\n",
+    "\n",
+    "    return summary_values"
```

Comment:
here you have defined 'result' as mean, max and min, but at the end you return 'summary_values'. So, summary_values is not deinfed and next code could not run